crates/sql/src/ast/mod.rs

```diff
@@ -0,0 +1,309 @@
+            },
+            [column] => SqlColumn {
+                column: column.clone().into(),
+                table: None,
+                alias: None,
+            },
+            _ => panic!("{}", &format!("Invalid column: `{value:?}`")),
+        }
+    }
+}
+
+impl fmt::Display for SqlColumn {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let alias = match &self.alias {
+            Some(alias) => format!(" AS {}", alias),
+            None => "".to_string(),
+        };
+        match &self.table {
+            Some(table) => write!(f, "{}.{}{alias}", table, self.column),
+            None => write!(f, "{}{alias}", self.column),
+        }
+    }
+}
+
+/// Helper to override the `Display` trait for `OpQuery` enum,
+/// to convert them to their `SQL` representation.
+#[derive(Debug)]
+pub struct PrintOperator {
+    pub(crate) op: OpQuery,
+}
+
+impl fmt::Display for PrintOperator {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self.op {
+            OpQuery::Cmp(op) => match op {
+                OpCmp::Eq => {
+                    write!(f, "=")
+                }
+                OpCmp::NotEq => {
+                    write!(f, "<>")
+                }
+                x => {
+                    write!(f, "{}", x)
+                }
+            },
+            OpQuery::Logic(op) => match op {
+                OpLogic::And => {
+                    write!(f, "AND")
+                }
+                OpLogic::Or => {
+                    write!(f, "OR")
+                }
+            },
+        }
+    }
+}
+
+/// Raw `SQL` value.
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub enum SqlValue {
+    /// Boolean value
+    Bool(bool),
+    /// Hexadecimal value
+    /// ```sql
+    /// 0x0000FCAB
+    /// x'93dda0'
+    /// ```
+    Hex(String),
+    /// Unparsed number (int, float, etc.)
+    Num(String),
+    /// Unparsed long number (bigint, etc.)
+    NumLong(String),
+    /// String value
+    String(String),
+    /// Identifier (table, column, etc.)
+    ///
+    /// ```sql
+    ///  "column_name"
+    /// ```
+    Identifier(Identifier),
+    Null,
+}
+
+impl fmt::Display for SqlValue {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            SqlValue::Bool(value) => write!(f, "{}", value),
+            SqlValue::Hex(value) => write!(f, "0x{}", value),
+            SqlValue::Num(value) => write!(f, "{}", value),
+            SqlValue::NumLong(value) => write!(f, "{}", value),
+            SqlValue::Identifier(value) => write!(f, "\"{}\"", value),
+            SqlValue::String(value) => write!(f, "'{}'", value),
+            SqlValue::Null => write!(f, "NULL"),
+        }
+    }
+}
+
+/// Field expression, with optional `alias`.
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Hash, From)]
+pub enum SqlFieldExpr {
+    Name(SqlColumn),
+    Value { value: SqlValue, alias: Option<Identifier> },
+}
```

Comment:
I think we want a field expression to just represent field access like `r.a`. In other words, `a` on its own should not be considered a field expression. And it shouldn't have an alias associated with it.